protocols/v2/framing-sv2/src/header.rs

```diff
@@ -69,25 +92,37 @@ impl Header {
         })
     }
 
-    /// Get the `Header` message type.
+    /// Get the [`Header`] message type.
     pub fn msg_type(&self) -> u8 {
         self.msg_type
     }
 
-    /// Get the `Header` extension type.
+    /// Get the [`Header`[ extension type.
     pub fn ext_type(&self) -> u16 {
         self.extension_type
     }
 
-    /// Check if `Header` represents a channel message
+    /// Check if [`Header`] represents a channel message.
     ///
     /// A header can represent a channel message if the MSB(Most Significant Bit) is set.
     pub fn channel_msg(&self) -> bool {
         const CHANNEL_MSG_MASK: u16 = 0b0000_0000_0000_0001;
         self.extension_type & CHANNEL_MSG_MASK == self.extension_type
     }
 
-    /// Calculate the length of the encrypted `Header`
+    /// Calculates the total length of a chunked message, accounting for MAC overhead.
+    ///
+    /// Determines the total length of the message frame, including the overhead introduced by
+    /// MACs. If the message is split into multiple chunks (due to its size exceeding the maximum
+    /// frame chunk size), each chunk requires a MAC for integrity verification.
+    ///
+    /// This method is particularly relevant when the message is being encrypted using the Noise
+    /// protocol, where the payload is divided into encrypted chunks, and each chunk is appended
+    /// with a MAC. However, it can also be applied to non-encrypted chunked messages to calculate
+    /// their total length.
```

Comment:
> However, it can also be applied to non-encrypted chunked messages to calculate their total length.

how? that sounds rather counter intuitive, given that the method is called `encrypted_len`

Reply:
I mean, I guess I can imagine that this is possible, since the `Header` object doesn't really know whether the frame has been encrypted or not (since that happens at a higher layer of abstraction)... so in theory yes, I guess the method can be called even if there's no encryption

but if there's no encryption and this method is called, it feels like the API is just being misused?

or to put my confusion in different words:

what is the use-case where it's useful to know the MAC overhead, while there's no MAC because there's no encryption?

Reply:
it seems this was pointed out on https://github.com/stratum-mining/stratum/pull/1179#issuecomment-2412645998

> Why don't we use this constant in the `Sv2Frame` decoder `WithoutNoise` struct? 

> To be clear, this technically does not have to do with Noise, but rather message chunking.

I think those assumptions are not true. This is related to Noise. See this constant:

https://github.com/stratum-mining/stratum/blob/384df6cd40394508d9345c956005f31a32dc200d/protocols/v2/const-sv2/src/lib.rs#L56-L57

And on [Noise Protocol Framework spec](https://noiseprotocol.org/noise.html#message-format), we have:
> All Noise messages are less than or equal to 65535 bytes in length.

So encryption imposes that large messages will have to be broken down into chunks.

Reply:
I also noticed that the previous documentation I wrote for this method was actually misleading.

> /// Calculate the length of the encrypted `Header`

that is not the case, since this will give the length of the encrypted payload, **not including the header**.

so thanks for taking the initiative to clarify this

Reply:
Yeah, I was confused by this method as you saw in my [comment](https://github.com/stratum-mining/stratum/pull/1179#issuecomment-2412645998). I figured my comments were wrong, but it would at least get a conversation started.

My confusion was indeed coming from the name of `SV2_FRAME_CHUNK_SIZE`, which is why I incorrectly stated in that comment "this technically does not have to do with Noise, but rather message chunking". 

I will remove my comments here and change to:

```rs
/// Calculate the length of the encrypted payload, not including the header.
```

I should have a better understanding of how the AEAD MAC works before expanding further on that comment to avoid putting incorrect data in. Perhaps we just leave that single sentence. If I do add anything else, I will tag you to bring your attention to it. And if you have any suggestions, just drop them here.

I added a suggestion to change the name to `NOISE_CHUNK_SIZE` [here](https://github.com/stratum-mining/stratum/issues/1135#issuecomment-2419824853) in the `const-sv2` refactor issue #1135.